### ISSUE TITLE
Posix.FileSys.wordToFD raises SysErr in callback to SML from C

### ISSUE THREAD

**Issue by pclayton** [polyml/polyml#132]
With Poly/ML 5.8, evaluating `Posix.FileSys.wordToFD fd` in an SML function that is called from C can raise the exception `SysErr ("Bad file descriptor", SOME EBADF)` if `fd` is derived outside the SML function but not raise an exception if `fd` is derived (in the same way) inside the SML function.  In either case, I would expect no exception to be raised.

This issue does not appear with earlier versions of Poly/ML, e.g. 5.7.1 is fine.

[This example](https://github.com/polyml/polyml/files/4791502/polyml_compiler_issue-20200617-1.tar.gz) demonstrates the issue by defining
`val null = Posix.FileSys.wordToFD 0w0`
outside the callback function and evaluating
`Posix.FileSys.fdToWord null`
in the callback function.

**Comment by dcjm** [comment]
I've pushed a fix (6aae4a8) for this.  It's actually an issue with `Posix.FileSys.wordToFD` and persistent file descriptors.  `null` is created when the code is compiled and exported but not used until the code is run.  It was actually possible to see the effect by defining `null`, calling `PolyML.SaveState.saveState` and then loading the state in a new session. 

By default, file descriptor values created in one session are turned into "bad descriptor" values when they are loaded into a new session since generally they are no longer valid.  That isn't correct for 0, 1 and 2 and persistent descriptors were added for `Posix.FileSys.stdin` etc.  I've changed `Posix.FileSys.wordToFD` so that it always creates persistent descriptors.
